### DISCUSSION TITLE
OLED on V3.2 vs V3.1

### DISCUSSION THREAD

**Discussion by tasrock** [ropg/heltec_esp32_lora_v3#69] [Q&A]
I wrote a program with V3.1 which utilized the OLED.  It worked great.  I just bought (2) new V3 boards.  (1) came in as V3.2 and (1) was V3.1.  I uploaded the same program onto the V3.2 board but now the OLED is so dim that I cannot read what it says even with the lights out.  I can tell that it is operating though.  Before I programmed it the demo was running and it looked good so I don't think the hardware is faulty.  I uploaded the same program to the new V3.1 and it worked just like my first V3.1 board.

I haven't seen anywhere where I can set the brightness or contrast of the OLED.  Am I just overlooking it?

Do you have any other thoughts on what might cause this?

**Comment by tasrock** [comment]
Update:
I just received (10) more of the V3.2 today.  I powered one up and it looked good with the factory demo.  I uploaded the same program as before and the display went dark again.  Just enough to tell that there was some text but not enough to read it.  I ran the situation by my AI overlord and it came up with a suggestion:

// Make sure Vext is properly configured in setup()
void setup() {
  // Turn on Vext
  pinMode(Vext, OUTPUT);
  digitalWrite(Vext, LOW);  // LOW turns on Vext
  delay(100);  // Give it time to stabilize
  
  display.init();
  display.setContrast(255);
  // ... rest of your setup code

I found that init and setContrast were already being called in heltec_setup().  I only added this code to my setup() after heltec_setup():
  pinMode(Vext, OUTPUT);
  digitalWrite(Vext, LOW);  // LOW turns on Vext
  delay(100);  // Give it time to stabilize
This seems to have fixed the issue.

It looks like heltec_ve() was handling this functionality already so now I am just calling heltec_ve(true) right after heltec_setup() in setup() function.

**Reply by Basitadam** [reply]
Worked for me, too. Thank you

**Reply by OneAndOnlyFinbar** [reply]
Can confirm this still works and is necessary to increase oled brightness.

**Reply by Digitalnorm** [reply]
Excellent. Works a treat for me too. Pity Heltec arent very forthcoming

**Reply by robingreig** [reply]
Worked for me as well. Thank You!

**Comment by peterbosch** [comment]
Brilliant (literally) - thank you! :-)

**Comment by ropg** [comment]
I'll make Vext on the default soon, so one has to turn it off explicitly.
